Vendor/iOS/MSAlertController/MSAlertController.m

```diff
@@ -146,4 +146,30 @@ + (void)presentNextAlertAnimated:(BOOL)animated {
   }
 }
 
+#define MS_DISPATCH_SELECTOR(result, class, selectorName, ...) ({ \
+  SEL selector = NSSelectorFromString(@#selectorName); \
+  IMP impl = [class methodForSelector:selector]; \
+  ((result (*)(id, SEL, ...)) impl)(class, selector, ##__VA_ARGS__); \
+})
+
++ (void)makeKeyAndVisible {
+  if (@available(iOS 13.0, *)) {
+    UIApplication *application = MS_DISPATCH_SELECTOR(UIApplication *, [UIApplication class], sharedApplication);
+    NSSet *scenes = MS_DISPATCH_SELECTOR(NSSet *, application, connectedScenes);
+    NSObject *windowScene = nil;
+    for (NSObject *scene in scenes) {
+      NSInteger activationState = MS_DISPATCH_SELECTOR(NSInteger, scene, activationState);
+      if (activationState == 0 /* UISceneActivationStateForegroundActive */) {
+        windowScene = scene;
+        break;
+      }
+    }
+    if (!windowScene) {
+      windowScene = scenes.anyObject;
```

Comment:
Are you sure that's not a broken state?

Reply:
https://github.com/microsoft/appcenter-sdk-apple/pull/1873#discussion_r345953000

Reply:
Taking any scene seems hazardous. I haven't looked deeply into this but it seems that observing the connected scenes seems to be the only reliable way to be notified of the next active window although quite cumbersome. Or maybe a way to intercept messages to scene delegates? Is there notifications to listen to, similar to the app lifecycle notifications?

Reply:
Well, I guess this is OK for **this** fix in the `Vendor/` utility folder, but as long term solution we need to make more changes not only in this alerts.

Based on [this](https://stackoverflow.com/a/56508769/1703290) good description:
> the UIApplication won't send background and foreground lifecycle messages to your app delegate. That means the code in these methods won't run:
> - `applicationDidBecomeActive`
> - `applicationWillResignActive`
> - `applicationDidEnterBackground`
> - `applicationWillEnterForeground`

So this means that most likely everywhere we use it (like session tracking, distribute, etc) we need some changes.

> Is there notifications to listen to, similar to the app lifecycle notifications?

Yes, `NSNotificationCenter` has `UISceneDidActivateNotification` for example that replaces application's one.

So, I guess we should have separate work item for:
- Adding new scenes to one of our app
- Refactor and centralize the work with app state somewhere in the core
- Probably raise the question to bump XCode version

Reply:
From what I could test this summer (would need to double check today) on the classic application lifecycle notifications still works even though scenes are implemented but of course it only captures life cycle of the application which is what we want so I am not very worried.

Reply:
This is a partial fix meaning it will fix the default 1 scene implementation provided by XCode but it won't support the multiple windows feature regardless of the number of scenes. For instance, beside the lifecycle cases there is another concern: in some cases we want the alert to be modal across all windows (mandatory update) but at this point it's broken. This is indeed bigger than just this PR we'll need design decisions and how to move forward. For now let's make the scope of this PR clear. It seems to be just fixing the "the default 1 scene implementation provided by XCode but not multiple windows apps".